x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/settings/AbstractSettingsService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.settings;
+
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.index.IndexNotFoundException;
+import org.elasticsearch.xpack.relevancesearch.settings.index.IndexCreationService;
+
+import java.util.Map;
+
+public abstract class AbstractSettingsService<S extends Settings> {
+    public static final String ENT_SEARCH_INDEX = ".ent-search";
+
+    protected final Client client;
+
+    public AbstractSettingsService(final Client client) {
+        this.client = client;
+    }
+
+    protected abstract String getName();
+
+    public S getSettings(String settingsId) throws SettingsNotFoundException, InvalidSettingsException {
```

Comment:
What is `S`?

Reply:
Nevermind, I saw `<S extends Settings>`.

Reply:
It's kind of confusing, as the[ convention in Java for type parameters](https://www.tutorialspoint.com/java_generics/java_generics_type_parameters.htm) is to use a single capital letter instead of using something like `Settings`.